c++/src/kj/compat/http.h

```diff
@@ -722,9 +758,30 @@ class HttpService {
   // Request processing can be canceled by dropping the returned promise. HttpServer may do so if
   // the client disconnects prematurely.
 
-  virtual kj::Promise<kj::Own<kj::AsyncIoStream>> connect(kj::StringPtr host);
-  // Handles CONNECT requests. Only relevant for proxy services. Default implementation throws
-  // UNIMPLEMENTED.
+  class ConnectTunnel {
+  public:
+    virtual kj::Own<kj::AsyncIoStream> accept(const HttpHeaders& headers) = 0;
```

Comment:
No status code or status text here because both are largely meaningless when establishing a tunnel. Any 2xx code is interpreted as success. Any other code means we're still operating under HTTP rules, which is addressed using the `reject()` method below (which does accept statusCode and statusText.

Reply:
What if a particular application wants to use different 2xx codes to signal something to the client?

Or for example in a proxying scenario, if the origin server returned 247 (made up number), I don't think we'd want the proxy to silently change the code to 200.

So I think we need to support specifying the status code here, unfortunately (and throw if it isn't 2xx).

Reply:
Ok. Not certain how likely that is but there's no harm in it.